### ISSUE TITLE
Support different naming formats

### ISSUE THREAD

**Issue by ThatNerdyPikachu** [jellyfin/jellyfin#1408]
**Describe the feature you'd like**

I'd like to be able to support different naming formats through plugins, or maybe just support for more common ones.

**Additional context**

An example of one commonly used (and Jellyfin doesn't detect / indentifies wrong) is <season>x<episode>, for example, 3x01, for season 3, episode 1

This actually might be due to how my GoT files are structured -- very weird how Jellyfin isn't picking it up tho..

```
TV Shows/
Game of Thrones S01 - S07 -jlw
     Game of Thrones S01/
         Game of Thrones - 1x01 - Winter Is Coming.mkv
```

The series isn't being recongnized as Game of Thrones automatically, and identifying it as GoT just shows every episode as the exact same one.

**Comment by SenorSmartyPants** [comment]
JF detects x based TV episodes, but you may need to have at least two digits for the episode. 

Example: show 3x01.mkv

**Comment by ThatNerdyPikachu** [comment]
See my additional context that I just added before seeing your comment.

**Comment by SenorSmartyPants** [comment]
This is a directory structure problem. 

https://github.com/MediaBrowser/Wiki/wiki/TV-naming

**Comment by ThatNerdyPikachu** [comment]
It's using the same S(xx) formatting as episodes use (minus the E(xx)), shouldn't that be supported as well, instead of the long form of Season (xx)?

**Comment by ThatNerdyPikachu** [comment]
This seems to already be implemented in a form, see https://github.com/jellyfin/jellyfin/blob/b4fdfb562dada77ff6d9c75bbc9e5e5b3ee977a2/Emby.Naming/TV/SeasonPathParser.cs#L81

It just doesn't currently see mine because of the Game of Thrones name before the S(xx)

**Comment by ThatNerdyPikachu** [comment]
@SenorSmartyPants This does seem to be partially implemented in a sense, as the seasons are being recongnized, just not the episodes.

**Comment by ThatNerdyPikachu** [comment]
Could I PR this?

**Comment by ThatNerdyPikachu** [comment]
As a sidenote, Plex supports this format out of the box, another reason for us to add it.

**Comment by k6Q28k5w5eWuRuuzTg9j7GrKCbjt7** [comment]
I tried to find some documentation from Emby 3.5.2 / when Jellyfin got forked to see if there were any more in depth information of supported naming schemes at that point. So far I have assumed [https://github.com/MediaBrowser/Wiki/wiki/TV-naming](https://github.com/MediaBrowser/Wiki/wiki/TV-naming) was supported (with less success). 

I think it could be very beneficial to come to a consensus, or recommendations, so users could organize data in a supported / to be supported manner (at least to some extent) as the documentation is very lacking in this regard. 

for example: 
` \TV Shows
     \A-M
       \Glee (2009)
          \Season 1
             Glee S01E01.mp4
     \N-Z
       \Seinfeld (1989)
          \Season 1
             Seinfeld S01E01.mp4`

Does not work on Jellyfin (each show gets identitified as a season of some random show based on the index. -And a season wont allow the user to select an Show id. 

tl;dr Vote to come to consensus on naming scheme(s) that should be supported and document them, even if not supported yet. Consensus on standard will benefit us all. I think :)
